### PR TITLE
Add IMM file handler

### PR DESCRIPTION
**Pull request by mrakitin**
This was done within the December Pilot meeting at APS/8-ID-I.

References:
- https://github.com/AdvancedPhotonSource/xpcs-eigen/blob/master/python/io/imm_file.py (original example)
- https://pypi.org/project/pyimm (another example)
- https://github.com/aps-8id-trr/ipython-8idiuser/blob/4a43a12a93460a578fd908f7f495811ee1150e79/profile_bluesky/startup/26-lambda.py#L268-L387 (implementation in the beamline profile)

### CODE REVIEW

File: area_detector_handlers/handlers.py [hunk 1]
```diff
@@ -317,3 +318,130 @@ def get_file_list(self, datum_kwargs_gen):
                 fn = self._template % (self._path, self._filename, j)
                 file_list.append(fn)
         return file_list
+
+
+def read_header(fp):
```
Comment (danielballan) [inline area_detector_handlers/handlers.py]: `fp` strongly suggests "filepath" to me, but in fact this is expected to be a file. Can we make the parameter name `file?
```suggestion
def read_header(file):
```

File: area_detector_handlers/handlers.py [hunk 2]
```diff
@@ -317,3 +318,130 @@ def get_file_list(self, datum_kwargs_gen):
...
+        'immversion',
+        'corecotick',
+        'cameratype',
+        'threshhold',
+        'byte632',
+        'empty_space',
+        'ZZZZ',
+        'FFFF'
+    ]
+    bindata = fp.read(1024)
```
Comment (danielballan) [inline area_detector_handlers/handlers.py]:
```suggestion
    bindata = file.read(1024)
```

File: area_detector_handlers/handlers.py [hunk 3]
```diff
@@ -317,3 +318,130 @@ def get_file_list(self, datum_kwargs_gen):
...
+
+    imm_headerdat = struct.unpack(imm_headformat, bindata)
+    imm_header ={}
```
Comment (danielballan) [inline area_detector_handlers/handlers.py]: This could be done as

```py
imm_header = dict(zip(imm_fieldnames, imm_headerdat))
```

Comment (mrakitin) [reply area_detector_handlers/handlers.py]: Implemented in 3f8cf7b.

**Comment by mrakitin** [comment]
Thanks, I agree with the approach.

**Pull request merged**